### PR TITLE
DM-11004: Create build-stack-docs command

### PR DESCRIPTION
**Pull request by jonathansick**
`build-stack-docs` runs documentation builds for stack projects like pipelines_lsst_io by:

1. Linking documentation content from packages into appropriate directories in the root project (`pipelines_lsst_io`).
2. Running Sphinx (via its Python API rather than creating a sub-process that runs `sphinx-build`).

To accomplish **1**, I've introduced a `manifest.yaml` file that packages should include in their `doc/` directories. This file declares what package and module documentation directories are included in the package so that they can be linked into the root project. There's an example of this file in the `tests/data/package_alpha/doc/manifest.yaml` file, and I've further documented it in an update to DMTN-030: https://dmtn-030.lsst.io/v/DM-11004/index.html#organization-of-documentation-in-packages

The implemented build algorithm is:

1. Discover packages according to eups setups.
2. Find documentation content by looking for `doc/manifest.yaml` files.
3. Link directories into the root project (`pipelines_lsst_io`).
4. Run the Sphinx build of the root project.

### CODE REVIEW

File: documenteer/stackdocs/build.py [hunk 1]
```diff
@@ -259,6 +266,11 @@ def find_package_docs(package_dir):
                 static_dirs=static_dirs)
 
 
+class NoPackageDocs(Exception):
+    """Exception raised when documentation is not found for an EUPS package.
+    """
```
Comment (SimonKrughoff) [inline documenteer/stackdocs/build.py]: I learned something today.  I though you would have to have a `pass` but declaration of a class only requires an indented block (the docstring in this case).

File: documenteer/stackdocs/build.py [hunk 2]
```diff
@@ -44,3 +49,128 @@ def discover_setup_packages():
...
+    package_dirs = {}
+    static_dirs = {}
+
+    if 'modules' in manifest_data:
+        for module_name in manifest_data['modules']:
+            module_dir = os.path.join(doc_dir, module_name)
+
+            # validate that the module's documentation directory does exist
+            if not os.path.isdir(module_dir):
+                continue
```
Comment (SimonKrughoff) [inline documenteer/stackdocs/build.py]: I'm a little confused by this.  If a module is in the manifest but the doc directory doesn't exist, shouldn't it complain a little louder than silently skipping?

Ignore this.  It is taken care of in a later commit.

File: documenteer/stackdocs/build.py [hunk 3]
```diff
@@ -44,3 +49,128 @@ def discover_setup_packages():
...
+    doc_dirs : namedtuple
+        Attributes of the named tuple are:
+
+        - ``package_dirs`` (`dict`). Keys are package names (for example,
+          ``'afw'``). Values are absolute directory paths to the package's
+          documentation directory inside the package's ``doc`` directory. If
+          there is no package-level documentation the dictionary will be empty.
+
+        - ``modules_dirs`` (`dict`). Keys are module names (for example,
+          ``'afw.table'``). Values are absolute directory paths to the module's
```
Comment (jonathansick) [inline documenteer/stackdocs/build.py]: Good catch, that was a mistake. My plan is to use full namespaces (with `lsst.`) just we end up with stacks, or have packages in the main stack, that don't have the common `lsst.` namespace.

File: tests/test_stackdocs_build.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+"""Tests for the documenteer.stackdocs.build module (build-stack-docs
+executable).
+"""
+
+import os
+
+from documenteer.stackdocs import build
+
+
+def test_find_package_docs():
```
Comment (jonathansick) [inline tests/test_stackdocs_build.py]: I'm running these tests with pytest (literally just run `pytest` from the package root).

File: documenteer/stackdocs/build.py [hunk 5]
```diff
@@ -44,3 +49,128 @@ def discover_setup_packages():
...
+          directory inside the package's ``doc`` directory. If a package has no
+          modules the returned dictionary will be empty.
```
Comment (jonathansick) [inline documenteer/stackdocs/build.py]: Yup, we're organizing content in `doc/` into a flat set of directories for Python modules and the package. I'm prototyping this out in `pipe_base` right now: https://github.com/lsst/pipe_base/tree/tickets/DM-11253/doc

**Pull request merged**